apps/src/music/views/MiniMusicPlayer.tsx

```diff
@@ -31,79 +33,98 @@ const MiniPlayerView: React.FunctionComponent<MiniPlayerViewProps> = ({
   );
   const sequencerRef = useRef<Simple2Sequencer>(new Simple2Sequencer());
   const sourcesStoreRef = useRef<SourcesStore>(new RemoteSourcesStore());
+  const analyticsReporter = useRef<AnalyticsReporter>(new AnalyticsReporter());
   const [isLoading, setIsLoading] = useState(true);
   const [currentProjectId, setCurrentProjectId] = useState<string | undefined>(
     undefined
   );
+  const {userId, userType, signInState} = useAppSelector(
+    state => state.currentUser
+  );
 
-  // Setup library and workspace on mount
+  // Setup library and workspace, and analyticsReporter on mount
   const onMount = useCallback(async () => {
     setUpBlocklyForMusicLab();
     workspaceRef.current.initHeadless();
     const library = await loadLibrary(libraryName);
     MusicLibrary.setCurrent(library);
     setIsLoading(false);
-  }, [libraryName]);
+    analyticsReporter.current.startSession();
```

Comment:
nit: I think we should use `await` here just to make sure the initialization completes before the function exits.